scipy/stats/tests/test_qmc.py

```diff
@@ -6,14 +6,242 @@
                            assert_equal, assert_array_almost_equal,
                            assert_array_equal)
 from scipy.stats import shapiro
+from scipy.optimize import basinhopping
 
 from scipy.stats._sobol import _test_find_index
 from scipy.stats import qmc
 from scipy.stats._qmc import (van_der_corput, n_primes, primes_from_2_to,
-                              _perturb_discrepancy, _update_discrepancy)
+                              _perturb_discrepancy, _update_discrepancy, QMCEngine)
+
+
+class QMCEngineTests:
```

Comment:
Yes, this looks like a good start. You'll move the Sobol tests up here?

Reply:
Yes, and I guess we actually don't need an extra test file just for Sobol'.

Reply:
@mdhaber I finished the refactoring of the doc, if you can have a look. I guess we could also adapt the remainings `MultinomialQMC`, `NormalQMC` and `MultivariateQMC` but I guess it's better to wait for you and @balandat to settle on the API, etc. first.

Reply:
Thanks, I'll take a closer look when I get to the QMCEngines.